src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -140,19 +184,47 @@ class DrawCall {
 
   // Insert an indexed indirect draw call into |pass|
   public drawIndexedIndirect(pass: GPURenderPassEncoder) {
+    // Generate index buffer
+    const indexArray = new Uint32Array(this.vertexCountInIndexBuffer).map((_, i) => i);
+    const indexBuffer = this.generateIndexBuffer(indexArray);
     this.bindVertexBuffers(pass);
-    pass.setIndexBuffer(this.indexBuffer, 'uint16');
+    pass.setIndexBuffer(indexBuffer, 'uint32');
     pass.drawIndexedIndirect(this.generateIndexedIndirectBuffer(), 0);
   }
 
   // Bind all vertex buffers generated
   private bindVertexBuffers(pass: GPURenderPassEncoder) {
     let currSlot = 0;
-    for (let i = 0; i < this.vertexBuffers.length; i++) {
+    // Deal with the instance step mode buffer
+    pass.setVertexBuffer(
+      currSlot++,
+      this.vertexBuffers[0],
+      !this.keepInstanceStepModeBufferInRange && this.offsetVertexBuffer ? 4 : 0
+    );
+    // Deal with the rest vertex step mode buffer
+    for (let i = 1; i < this.vertexBuffers.length; i++) {
       pass.setVertexBuffer(currSlot++, this.vertexBuffers[i], this.offsetVertexBuffer ? 4 : 0);
     }
   }
 
+  // Create a buffer with given size and usage: usage | GPUBufferUsage.COPY_DST, and call
+  // writeBuffer with given data and parameter.
+  private createBufferWithContent(
```

Comment:
I see, but currently GPUTest.makeBufferWithContents use the byte length of given data array as the size of buffer, but in case of `partialLastNumber` we have a size of Float32Array minus 1 byte, which means 4n+3 bytes, and makeBufferWithContents can not do this. In this case we also can not use `mappedAtCreation: true`.

Reply:
@kainino0x would it be fine to fix `makeBufferWithContents` to round the size up to 4?

Reply:
Yes, it would be fine. There must not be any tests doing that right now as it would be causing a validation error if there were.

Reply:
Wait, both createBuffer and writeBuffer require the size to be a multiple of 4 bytes. How would rounding up the size help?

Reply:
Yes, in this test we want to create a strange vertex buffer of size `(4n+3)` bytes, and write `4n` bytes of data into this buffer, while dataArray actually have `(4n+4)` bytes data. Using makeBufferWithContents we can only create buffers with `4n` bytes, but using createBuffer we can have that odd size of buffer.

Reply:
Oh, sorry, I misread the spec. Buffers only have to be a multiple of 4 bytes in size if they are mappedAtCreation.

Reply:
In that case I think it's best to just not use makeBufferWithContents.

Alternative would be to have e.g. makeBufferWithContents use writeBuffer if not a multiple of 4 bytes, but that's too much of a special case I think.